core/sourcehook/generate/sourcehook.hxx

```diff
@@ -566,6 +570,730 @@ namespace SourceHook
+		struct HookHandlerImpl
+		{
+		public:
+			/**
+			 * @brief The delegate type that SourceHook will invoke (user code)
+			 */
+			typedef typename fastdelegate::FastDelegate<Result, Args...> Delegate;
+
+			struct IMyDelegate : ::SourceHook::ISHDelegate
+			{
+			public:
+				virtual Result Call(Args... args) = 0;
+			};
+
+			struct CMyDelegateImpl : IMyDelegate
+			{
+			public:
+				Delegate _delegate;
+
+				CMyDelegateImpl(Delegate deleg) : _delegate(deleg)
+				{}
+				virtual~CMyDelegateImpl() {}
+				Result Call(Args... args) { return _delegate(args...); }
+				void DeleteThis() { delete this; }
+				bool IsEqual(ISHDelegate *pOtherDeleg) { return _delegate == static_cast<CMyDelegateImpl *>(pOtherDeleg)->_delegate; }
+
+			};
+
+			/**
+			*	@brief An object containing a safely-allocatable return type
+			*
+			*	Used when stack-allocating the return type;
+			*	void returns are never written to so using void* is safe.
+			*	when the return is not zero-sized, use the return itself.
+			*/
+			typedef typename metaprogramming::if_else<
+					std::is_void<Result>::value,
+					void*,
+					Result
+			>::type VoidSafeResult;
+
+			/**
+			*	@brief A return type that is C++-reference-safe.
+			*
+			*	Prevents us from doing silly things with byref-passed values.
+			*/
+			typedef typename ReferenceCarrier<VoidSafeResult>::type ResultType;
+
+			::SourceHook::ProtoInfo Proto;
+
+			/**
+			*	@brief Build the ProtoInfo for this hook
+			*/
+			constexpr HookHandlerImpl()
+			{
+				//	build protoinfo
+				Proto.numOfParams = sizeof...(Args);
+				Proto.convention = Convention;
+
+				if constexpr (std::is_void_v<Result>) {
+					Proto.retPassInfo = {0, 0, 0};
+				} else {
+					Proto.retPassInfo = { sizeof(Result), ::SourceHook::GetPassInfo< Result >::type, ::SourceHook::GetPassInfo< Result >::flags };
+				}
+
+				//	Iterate over the args... type pack
+				auto paramsPassInfo = new PassInfo[sizeof...(Args) + 1];
+				paramsPassInfo[0] = { 1, 0, 0 };
+				Proto.paramsPassInfo = paramsPassInfo;
+
+				detail::PrototypeBuilderFunctor argsBuilder(paramsPassInfo);
+				metaprogramming::for_each_template_nullable<detail::PrototypeBuilderFunctor, Args...>(&argsBuilder);
+
+
+				//	Build the backwards compatible paramsPassInfoV2 field
+				Proto.retPassInfo2 = {0, 0, 0, 0};
+				auto paramsPassInfo2 = new PassInfo::V2Info[sizeof...(Args) + 1];
+				Proto.paramsPassInfo2 = paramsPassInfo2;
+
+				for (int i = 0; i /* lte to include thisptr! */ <= sizeof...(Args); ++i) {
+					paramsPassInfo2[i] = { 0, 0, 0, 0 };
+				}
+
+			}
+
+			template<ISourceHook** SH, typename Invoker, typename InstType>
+			static Result HookImplCore(InstType* Instance, void* self, Args... args)
+			{
+				//	Note to all ye who enter here:
+				//	Do not, do NOT--DO NOT: touch "this" or any of our member variables.
+				//	Hands off bucko. USE THE STATIC INSTANCE! (thisptr is undefined!!)
```

Comment:
That comment should be unnecessary, given that the function is static, `this` keyword is out of bounds.

Reply:
Ah yes that was inherited from when that method used to be virtual. Will smite that, thanks